src/imageupload/imageuploadediting.js

```diff
@@ -81,6 +93,54 @@ export default class ImageUploadEditing extends Plugin {
 			} );
 		} );
 
+		// Handle HTML pasted with images with base64 or blob sources.
+		// For every image file, a new file loader is created and a placeholder image is
+		// inserted into the content. Then, those images are uploaded once they appear in the model
+		// (see Document#change listener below).
+		this.listenTo( editor.plugins.get( 'Clipboard' ), 'inputTransformation', ( evt, data ) => {
+			const view = editor.editing.view;
+
+			const fetchableImages = Array.from( view.createRangeIn( data.content ) )
+				.filter( value => isLocalImage( value.item ) && !value.item.getAttribute( 'uploadProcessed' ) )
+				.map( ( value, index ) => wrapImageToFetch( value.item, index ) );
+
+			if ( !fetchableImages.length ) {
+				return;
+			}
+
+			evt.stop();
+
+			Promise.all( fetchableImages ).then( items => {
+				for ( const item of items ) {
+					if ( !item.file ) {
+						// Failed to fetch image or create a file instance, remove image element.
+						view.change( writer => {
+							writer.remove( item.image );
+						} );
+					} else {
+						const loader = fileRepository.createLoader( item.file );
+
+						if ( loader ) {
+							view.change( writer => {
+								writer.setAttribute( 'src', '', item.image );
+								writer.setAttribute( 'uploadId', loader.id, item.image );
+							} );
+						} else {
+							view.change( writer => {
+								// Set attribute so the image will not be processed 2nd time.
+								writer.setAttribute( 'uploadProcessed', true, item.image );
```

Comment:
I'd move this out of this condition. It would be more intuitive to read if we marked all processed images. Not only half of them. After all, we don't want to process again all images that we already processed.